### ISSUE TITLE
Spring 4.3.1 errors on latest master: ClassNotFoundException: org.springframework.core.io.Resource

### ISSUE THREAD

**Issue by pearj** [HotswapProjects/HotswapAgent#140]
I have build the latest hotswap agent, as I noticed some changes for Spring 4.x, however I've noticed that I get this error on some changes when using Spring 4.3.1

```
23:44:04,264 ERROR [stderr] (Thread-270) Exception in thread "Thread-270" java.lang.NoClassDefFoundError: org/springframework/core/io/Resource
23:44:04,267 ERROR [stderr] (Thread-270)        at java.lang.Class.getDeclaredMethods0(Native Method)
23:44:04,268 ERROR [stderr] (Thread-270)        at java.lang.Class.privateGetDeclaredMethods(Class.java:2625)
23:44:04,272 ERROR [stderr] (Thread-270)        at java.lang.Class.getDeclaredMethod(Class.java:2015)
23:44:04,273 ERROR [stderr] (Thread-270)        at org.hotswap.agent.plugin.spring.scanner.ClassPathBeanRefreshCommand.executeCommand(ClassPathBeanRefreshCommand.java:72)
23:44:04,274 ERROR [stderr] (Thread-270)        at org.hotswap.agent.command.impl.CommandExecutor.run(CommandExecutor.java:25)
23:44:04,278 ERROR [stderr] (Thread-270) Caused by: java.lang.ClassNotFoundException: org.springframework.core.io.Resource
23:44:04,279 ERROR [stderr] (Thread-270)        at java.net.URLClassLoader$1.run(URLClassLoader.java:366)
23:44:04,280 ERROR [stderr] (Thread-270)        at java.net.URLClassLoader$1.run(URLClassLoader.java:355)
23:44:04,281 ERROR [stderr] (Thread-270)        at java.security.AccessController.doPrivileged(Native Method)
23:44:04,283 ERROR [stderr] (Thread-270)        at java.net.URLClassLoader.findClass(URLClassLoader.java:354)
23:44:04,284 ERROR [stderr] (Thread-270)        at java.lang.ClassLoader.loadClass(ClassLoader.java:425)
23:44:04,285 ERROR [stderr] (Thread-270)        at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:308)
23:44:04,287 ERROR [stderr] (Thread-270)        at java.lang.ClassLoader.loadClass(ClassLoader.java:358)
23:44:04,288 ERROR [stderr] (Thread-270)        ... 5 more
```

What causes this? Is this error expected? Should HotswapAgent be able to reprocess annotations?

**Comment by edudant** [comment]
This should work in current version. Please reopen if still valid.

**Comment by sachinkakkar7** [comment]
Hi,
Can we please re-open this issue? This issue is still happening with latest hotswap agent jar.

**Comment by skybber** [comment]
Have you this problem on Wildfly or on another appserver?

**Comment by sachinkakkar7** [comment]
I am having this problem with JBoss Application Server 7.

**Comment by skybber** [comment]
Is the error message same like in #184 ?

**Comment by sachinkakkar7** [comment]
That's correct. Its similar to #184

**Comment by skybber** [comment]
fixed in snapshot 1.1.0-SNAPHOT
